includes/sanitizers/class-amp-iframe-sanitizer.php

```diff
@@ -304,6 +305,13 @@ private function normalize_attributes( $attributes ) {
 					// No need to copy.
 					break;
 
+				case 'type':
+					/*
+					 * Though type is non-standard attribute in <iframe> tag, some embeds (i.e Amazon Kindle Embed) uses
+					 * it, therefore we need to omit this.
+					 */
+					break;
+
```

Comment:
```suggestion
				case 'type':
					/*
					 * Omit the `type` attribute if its value is `text/html`. Popular embed providers such as YouTube
					 * and Amazon use this non-standard attribute.
					 */
					if ( 'text/html' !== strtolower( $value ) ) {
						$out[ $name ] = $value;
					}
					break;
```

Reply:
Also, the YouTube embed code doesn't use this `type` attribute as far as I know. It's only the Jetpack embed handler for YouTube that used to use it (but now does no longer).

Reply:
Oh I see. I dug further into this, and I believe the `type` attribute may be a remnant of when the YouTube API used to return `<object>` and `<embed>` tags, before the switch to [iframes in 2011](https://youtube-eng.googleblog.com/2011/05/youtube-google-io-iframe-player-api_31.html). [This issue](https://bugzilla.mozilla.org/show_bug.cgi?id=769117) reported on Bugzilla does refer to the `<object>` embed having the `type` attribute set to `text/html`, for example.

Fun fact: you can still get `<object>` embeds by setting the `iframe` query argument to `0` in an oEmbed URL.